### PR TITLE
Support VIP

### PR DESCRIPTION
**Pull request by yaocw2020**
Provide a VIP for the Harvester.

**Related Issue:**
https://github.com/harvester/harvester/issues/632

### CODE REVIEW

File: deploy/charts/harvester/templates/service.yaml [hunk 1]
```diff
@@ -6,6 +6,13 @@ metadata:
   # name: {{ template "harvester.fullname" . }}
   # because we can easily confirm this resource from the corresponding namespace.
   name: harvester
+  annotations:
+{{- if and (eq .Values.service.harvester.type "LoadBalancer") (eq .Values.service.harvester.vipMode "DHCP") }}
+{{- if and (ne .Values.service.harvester.vip "") (ne .Values.service.harvester.vipHwAddr "") }}
```
Comment (gitlawr) [inline deploy/charts/harvester/templates/service.yaml]: a non-empty string is evaluated to be true. So we don't need the ne checks.

File: deploy/charts/harvester/values.yaml [hunk 2]
```diff
@@ -221,7 +220,7 @@ service:
     ## Specify the port of HTTPs endpoint.
     ## defaults to "8443".
     ##
-    httpsPort: 8443
+    httpsPort: 443
```
Comment (gitlawr) [inline deploy/charts/harvester/values.yaml:223]: Have we changed the port?

Comment (yaocw2020) [reply deploy/charts/harvester/values.yaml:223]: Yes. The default port of HTTPS is 443.

Comment (gitlawr) [reply deploy/charts/harvester/values.yaml:223]: Also, update the default value in the comment

File: deploy/charts/harvester/values.yaml [hunk 3]
```diff
@@ -377,3 +387,8 @@ webhook:
     tag: master-head
 
   httpsPort: 9443
+
+kube-vip:
+  enabled: true
+  config:
+      vip_interface: eth0
```
Comment (bk201) [inline deploy/charts/harvester/values.yaml]: Does this mean VIP is always on `eth0`?

Comment (yaocw2020) [reply deploy/charts/harvester/values.yaml]: It's the default value. Once users enable the VLAN network and use eth0 as the VLAN NIC, eth0 is unavailable for the VIP interface and we will switch the VIP interface as `harvester-br0`.

Comment (gitlawr) [reply deploy/charts/harvester/values.yaml]: I don't see we override it in installation. Should we set it to be the management NIC?

**Review by gitlawr — APPROVED** [review]
LGTM

File: deploy/charts/harvester/values.yaml [hunk 4]
```diff
@@ -377,3 +387,8 @@ webhook:
     tag: master-head
 
   httpsPort: 9443
+
+kube-vip:
```
Comment (guangbochen) [inline deploy/charts/harvester/values.yaml:414]: will need to add the image of the kube-vip, kube-vip-cloud-provider, and harvester-load-balancer respectively.

**Review by guangbochen — APPROVED** [review]
LGTM

**Review by bk201 — APPROVED** [review]
I can confirm the images are also pre-loaded in the installer side.

**Pull request merged**